tests/common.go

```diff
@@ -12936,3 +12953,128 @@ func GetMultipathDeviceOnPool(n *node.Node) (map[string][]string, error) {
 	}
 	return multipathMap, nil
 }
+
+// RunCmdsOnAllMasterNodes Runs a set of commends on all the master nodes
+func RunCmdsOnAllMasterNodes(cmds []string) error {
+	for _, node := range node.GetMasterNodes() {
+		for _, cmd := range cmds {
+			log.InfoD(fmt.Sprintf("Running command %s on %s", cmd, node))
+			_, err := runCmdOnceNonRoot(cmd, node)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ConfigureClusterLevelPSA Configure cluster level PSA settings where all newly created namespaces will be affected by
+// it and will exclude kube-system, default, px-backup and portworx namespace
+func ConfigureClusterLevelPSA(psaProfile string) error {
+
+	// Get the namespace where portworx is present
+	pxNs, err := Inst().S.GetPortworxNamespace()
+	if err != nil {
+		return err
+	}
+
+	// Get the namespace where px-backup is present
+	pxBackupNamespace, err := backup.GetPxBackupNamespace()
+	if err != nil {
+		return err
+	}
+
+	// Create a list of all the namespaces which need to be excluded
+	namespaces := []string{"default", "kube-system", pxBackupNamespace}
+	if pxNs != "kube-system" {
+		namespaces = append(namespaces, pxNs)
+	}
+	joined := "\"" + strings.Join(namespaces, "\",\"") + "\""
+
+	tempFilePath := kubeApiServerConfigFilePath + ".tmp"
+
+	cmds := []string{
+		fmt.Sprintf("mkdir -p /etc/kubernetes/admission"),
+		fmt.Sprintf("curl -o %s http://kubevirt-disk-registry.pwx.dev.purestorage.com/more_images/admissioncontroller.yaml", KubeAdmissionControllerFilePath),
+		fmt.Sprintf("sed -i 's/{Profile}/%s/' %s", psaProfile, KubeAdmissionControllerFilePath),
+		fmt.Sprintf("sed -i 's/{NS}/%s/' %s", joined, KubeAdmissionControllerFilePath),
+		fmt.Sprintf("cat  %s > %s", kubeApiServerConfigFilePath, tempFilePath),
+		fmt.Sprintf(`sed -i -e '/- kube-apiserver/a\ \ \  - --admission-control-config-file=/etc/kubernetes/admission/admissioncontroller.yaml' %s`, tempFilePath),
+		fmt.Sprintf(`sed -i -e '/volumeMounts:/a\ \ \  - mountPath: /etc/kubernetes/admission/\n\ \ \ \ \ \ name: admission-conf\n\ \ \ \ \ \ readOnly: true' %s`, tempFilePath),
+		fmt.Sprintf(`sed -i -e '/volumes:/a\  - hostPath:\n\ \ \ \ \ \ path: /etc/kubernetes/admission/\n\ \ \ \ \ \ type: DirectoryOrCreate\n\ \ \ \ name: admission-conf' %s`, tempFilePath),
+		fmt.Sprintf("cat  %s > %s", kubeApiServerConfigFilePath, kubeApiServerConfigFilePathBkp),
+		fmt.Sprintf("mv %s %s", tempFilePath, kubeApiServerConfigFilePath),
+	}
+	log.Infof(fmt.Sprintf("%s", strings.Join(cmds, "\n")))
+	// Run the above set of commands in all the master nodes
+	err = RunCmdsOnAllMasterNodes(cmds)
+	if err != nil {
+		return err
+	}
+
+	time.Sleep(2 * time.Minute)
```

Comment:
Why are we adding sleep when we are using DoRetryWithTimeout?

Reply:
Done